src/d/actor/d_a_tag_etc.cpp

```diff
@@ -4,50 +4,175 @@
 //
 
 #include "d/actor/d_a_tag_etc.h"
+#include "d/d_com_inf_game.h"
+#include "d/d_event.h"
+#include "d/d_event_manager.h"
 #include "dolphin/types.h"
+#include "f_op/f_op_actor_iter.h"
+#include "f_pc/f_pc_searcher.h"
+#include "global.h"
 
 /* 00000078-00000084       .text getEventNo__11daTag_Etc_cFv */
-void daTag_Etc_c::getEventNo() {
-    /* Nonmatching */
+u8 daTag_Etc_c::getEventNo() {
+    return this->mBase.mParameters >> 0x18;
 }
 
 /* 00000084-00000090       .text getType2__11daTag_Etc_cFv */
-void daTag_Etc_c::getType2() {
-    /* Nonmatching */
+u32 daTag_Etc_c::getType2() {
+    return this->mBase.mParameters >> 8 & 0xF;
 }
 
+/* Not Matching */
 /* 00000090-000001B4       .text rangeCheck__11daTag_Etc_cFP10fopAc_ac_c */
-void daTag_Etc_c::rangeCheck(fopAc_ac_c*) {
-    /* Nonmatching */
+BOOL daTag_Etc_c::rangeCheck(fopAc_ac_c* pActor) {
+    BOOL result;
+    float fVar4;
+    float distanceMagnitude;
+    cXyz cxYz_planar_distance;
+    cXyz cXyz_distance;
+
+    cXyz_distance = ((pActor->current).pos - (this->current).pos);
+    if (0.0 <= cXyz_distance.x) {
+        cxYz_planar_distance.x = cXyz_distance.x;
+        cxYz_planar_distance.y = 0.0;
+        cxYz_planar_distance.z = cXyz_distance.z;
+        distanceMagnitude = PSVECSquareMag(&cxYz_planar_distance);
+        if (0.0 < distanceMagnitude) {
+            fVar4 = sqrtf(distanceMagnitude);
+        }
+        distanceMagnitude = this->mScale.x;
+        distanceMagnitude *= 100;
+        if ((distanceMagnitude <= fVar4) || ((this->mScale.x) * 100 < cXyz_distance.y)) {
+            result = FALSE;
+        } else {
+            result = TRUE;
+        }
+    } else {
+        result = FALSE;
+    }
+    return result;
 }
 
 /* 000001B4-00000214       .text otherCheck__11daTag_Etc_cFP10fopAc_ac_c */
-void daTag_Etc_c::otherCheck(fopAc_ac_c*) {
-    /* Nonmatching */
-}
-
+BOOL daTag_Etc_c::otherCheck(fopAc_ac_c* pActor) {
+    BOOL result;
+
+    u8 cVar2 = getType2();
+
+    switch (cVar2) {
+    case 0:
+        if (pActor != (fopAc_ac_c*)0x0 &&
+            ((u32)pActor[0x13].mBase.mLnTg.mBase.mpTagData & 0x10) != 0)
+        {
+            result = TRUE;
+        } else {
+            result = FALSE;
+        }
+        break;
+    default:
+        result = TRUE;
+        break;
+    }
+    return result;
+}
+
+/* Not matching, registers seem off */
 /* 00000214-000002EC       .text demoProc__11daTag_Etc_cFv */
 void daTag_Etc_c::demoProc() {
-    /* Nonmatching */
+    s16 sVar1;
+    fopAc_ac_c* pfVar2;
+    int staffIdx;
+    u8 cVar3;
+    void* local_18[2];
+
+    local_18[0] = this->field_0x294;
+    pfVar2 = (fopAc_ac_c*)fopAcIt_Judge(fpcSch_JudgeByID, local_18);
+    staffIdx = g_dComIfG_gameInfo.play.mEvtManager.getMyStaffId("TAG_ETC_D", (fopAc_ac_c*)0x0, 0);
```

Comment:
```suggestion
    staffIdx = dComIfGp_evmng_getMyStaffId("TAG_ETC_D");
```

actors basically never access g_dComIfG_gameInfo directly, you should always use the appropriate dComIfG inline (which you can see in the demo debug maps) unless it doesn't match. same for several other calls like this in this file.